src/core/xconnection.rs

```diff
@@ -520,13 +522,16 @@ impl XcbConnection {
         // xcb docs: https://www.mankier.com/3/xcb_randr_select_input
         xcb::randr::select_input(&conn, root, NOTIFY_MASK).request_check()?;
 
+        let modifier_map = modifiers_from_xmodmap();
```

Comment:
I'd prefer for the modifiers to be passed through in the parameters to `new` so that we aren't hard coding a dependency on `xmodmap` into `XcbConnection`. The `gen_keybindings` macro is called outside of the constructor for `WindowManager` for the same reason: it allows users to generate the map in an alternate way if they need/want to.

Reply:
That makes sense.